### ISSUE TITLE
Hide all resources except admin users 

### ISSUE THREAD

**Issue by pedami** [kirschbaum-development/nova-mail#57]
Hello, we need to hide all the resources of this package from non-admin users, that would be great.
Thanks.

**Comment by adammparker** [comment]
Hello @pedami, you should be able to accomplish that with model policies. You can also override the Nova resources with your own versions, this is covered in the README here: https://github.com/kirschbaum-development/nova-mail#resource-customization

**Comment by pedami** [comment]
Hello again, could you give me an example?
I have created a Resoruce NovaSentMailCustom.php and a Policy NovaSentMailCustomPolicy.php but I can't protect it for non-admin users unless I go to your vendor namespace KirschbaumDevelopment\NovaMail\Policies\ NovaSentMailPolicy and add this couple of lines:

public function viewAny(User $user): bool {
         return false;
     }


I have tried to put them in my code, but there is no way:

In addition, the other two sections of the menu do not have Policy, templates, or events.


**This is my resource**


<?php

namespace App\Nova;

use Illuminate\Http\Request;
use KirschbaumDevelopment\NovaMail\Nova\NovaSentMail;

class NovaSentMailCustom extends NovaSentMail
{

    public function cards(Request $request)
{
    return [
        // Your custom code...
    ];
}
}

<img width="600" alt="image" src="https://user-images.githubusercontent.com/6138775/184886049-c541b62f-5f90-4088-aead-379a44ebb5f3.png">


**And this is my policy**
<?php

namespace App\Policies;

use App\Models\User;
use Illuminate\Auth\Access\HandlesAuthorization;
use Illuminate\Contracts\Auth\Authenticatable;
use App\Models\NovaSentMail;


class NovaSentMailCustomPolicy extends NovaSentMailPolicy
{
    use HandlesAuthorization;


    // Ocultar a todos los usuarios
    public function viewAny(User $user): bool {
        return false;
    }

}
<img width="600" alt="image" src="https://user-images.githubusercontent.com/6138775/184885956-79986d6a-2cc7-4f22-a0a8-5c61fa84ff07.png">

**Comment by pedami** [comment]
<?php

namespace App\Nova;

use Illuminate\Http\Request;
use KirschbaumDevelopment\NovaMail\Nova\NovaSentMail;

class NovaSentMailCustom extends NovaSentMail
{

    public function cards(Request $request)
{
    return [
        // Your custom code...
    ];
}
}
?>


<?php

namespace App\Policies;

use App\Models\User;
use Illuminate\Auth\Access\HandlesAuthorization;
use Illuminate\Contracts\Auth\Authenticatable;
use App\Models\NovaSentMail;


class NovaSentMailCustomPolicy extends NovaSentMailPolicy
{
    use HandlesAuthorization;


    // Ocultar a todos los usuarios
    public function viewAny(User $user): bool {
        return false;
    }

}
?>

**Comment by adammparker** [comment]
Hello @pedami, I just did some additional testing here and I was able to get a custom policy to work.

First I created `App\Policies\NovaSentMailPolicy` (name doesn't matter) with a `viewAny` method that returned `false`.

In `AuthServiceProvider` after the `registerPolicies()` call..
```php
use App\Policies\NovaSentMailPolicy;
use KirschbaumDevelopment\NovaMail\Models\NovaSentMail;

...

Gate::policy(NovaSentMail::class, NovaSentMailPolicy::class);
```

I also tested this with a custom resource that overrides the `NovaSentMail` resource, that doesn't really change anything since the policy and policy registration are based on the model, not the resource. If you have a custom model (and custom resource by extension) I believe this would still work, although I did not test that far.

Even though the package does not include a policy class for `NovaMailTemplate` or `NovaMailEvent` it also works there. I tested this by creating `App\Policies\NovaMailTemplatePolicy`.

Then, just like above, I did the same thing in the application's `AuthServiceProvider` as above, I just registered the policy with the `NovaMailTemplate` model.

```php
use App\Policies\NovaMailTemplatePolicy;
use KirschbaumDevelopment\NovaMail\Models\NovaMailTemplate;

...

Gate::policy(NovaMailTemplate::class, NovaMailTemplatePolicy::class);
```
